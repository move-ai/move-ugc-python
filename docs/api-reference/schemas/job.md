::: move_ugc.schemas.job.JobType
    options:
        members: ["id", "created", "metadata", "client", "outputs", "state", "take"]

::: move_ugc.schemas.job.JobOptions