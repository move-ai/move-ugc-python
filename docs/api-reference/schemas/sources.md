# Sources

::: move_ugc.schemas.sources.TakeSourceKey
    options:
        members:
            - mp4
            - avi
            - mov
            - move

::: move_ugc.schemas.sources.CameraSettings
    options:
        members:
            - lens

::: move_ugc.schemas.sources.AdditionalFileType
    options:
        members:
            - key
            - file


::: move_ugc.schemas.sources.SourceIn
    options:
        members:
            - device_label
            - file_id
            - format
            - camera_settings

::: move_ugc.schemas.sources.Source
    options:
        members:
            - device_label
            - file
            - format
            - camera_settings
