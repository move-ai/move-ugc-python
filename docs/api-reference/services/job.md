# Jobs

::: move_ugc.services.job.JobService
    options:
        members:
            - create
            - retrieve
            - update
            - list
