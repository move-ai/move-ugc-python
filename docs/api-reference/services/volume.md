# Volume

::: move_ugc.services.take.VolumeService
    options:
        members:
            - create_human_volume
            - retrieve_human_volume