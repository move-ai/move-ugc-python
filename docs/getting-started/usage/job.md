Jobs are the processing entity in MoveUGC. By creating a Job, you can initiate the processing for a take. 
A job can have multiple output types, currently the only output type supported is `mp4` and `fbx` files.

## Prequisites

Before creating a job, please make sure that a take is created. If you've not created a take please refer to [this usage guide](/move-ugc-python/latest/getting-started/usage/take/).

## Creating a job

```python
job = ugc.jobs.create(take_id="take-2be2463e-ffa3-419b-beb4-ea0f99c79512")
```

## Attaching some custom metadata with your job

The metadata attribute in job type accepts any valid json string and can contain any custom data. This is particularly useful if any business logic needs to be implemented such as attaching a user id to a job.

```python
take = ugc.jobs.create(
    take_id="take-2be2463e-ffa3-419b-beb4-ea0f99c79512",
    metadata=json.dumps({"foo": "bar"}),
)
```