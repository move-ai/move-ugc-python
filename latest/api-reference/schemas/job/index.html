<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <title>Redirecting</title>
  <noscript>
    <meta http-equiv="refresh" content="1; url=../../../../0.7.12/api-reference/schemas/job/" />
  </noscript>
  <script>
    window.location.replace("../../../../0.7.12/api-reference/schemas/job/" + window.location.hash);
  </script>
</head>
<body>
  Redirecting to <a href="../../../../0.7.12/api-reference/schemas/job/">../../../../0.7.12/api-reference/schemas/job/</a>...
</body>
</html>