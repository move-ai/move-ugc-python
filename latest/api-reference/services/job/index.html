<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <title>Redirecting</title>
  <noscript>
    <meta http-equiv="refresh" content="1; url=../../../../3.3.1/api-reference/services/job/" />
  </noscript>
  <script>
    window.location.replace("../../../../3.3.1/api-reference/services/job/" + window.location.hash);
  </script>
</head>
<body>
  Redirecting to <a href="../../../../3.3.1/api-reference/services/job/">../../../../3.3.1/api-reference/services/job/</a>...
</body>
</html>