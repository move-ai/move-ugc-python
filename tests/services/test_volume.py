"""Unit tests for using the volume service."""

import pytest
from graphql import ExecutionResult
from pydantic import ValidationError

from move_ugc.schemas.constants import OUTPUTS_LITERAL
from move_ugc.schemas.job import ClipWindow
from move_ugc.schemas.sources import SourceIn, TakeSourceKey
from move_ugc.schemas.volume import AreaType
from tests.constants import LIST_VOLUMES_QUERY
from tests.services.testcases import ServicesTestCase

SOURCES_LITERAL = "sources"


class TestVolumeService(ServicesTestCase):
    """Test volume service."""

    service_name = "volumes"

    @pytest.mark.parametrize(
        argnames="expand, take_fixture",
        argvalues=[
            (None, "volume_create_response"),
            ([], "volume_create_response"),
            (["client"], "volume_create_response_with_client"),
            ([SOURCES_LITERAL], "volume_create_response_with_video_source"),
            ([SOURCES_LITERAL], "volume_create_res_with_additional_sources"),
            (["clipWindow"], "volume_create_response_with_clip_window"),
        ],
        ids=[
            "no_expand",
            "empty_expand",
            "expand_client",
            "expand_video_file",
            "expand_additional_sources",
            "expand_clip_window",
        ],
    )
    def test_create(  # noqa: WPS211
        self,
        snapshot,
        snapshot_json,
        faker,
        request,
        expand,
        take_fixture,
    ):
        """Test creating a take.

        This should test -> `ugc.volumes.create_human_volume()`

        Args:
            snapshot: The snapshot fixture.
            snapshot_json: The snapshot json fixture.
            faker: The faker fixture.
            request: The request fixture.
            expand: The expand fixture.
            take_fixture: The take fixture.
        """
        request.getfixturevalue(take_fixture)
        device_label = faker.word()
        volume = self.client.volumes.create_human_volume(
            metadata=request.getfixturevalue("metadata_for_update"),
            clip_window=ClipWindow(
                start_time=faker.pyint(),
                end_time=faker.pyint(),
            ),
            sources=[
                SourceIn(
                    device_label=device_label,
                    file_id=faker.uuid4(),
                    format=TakeSourceKey.mp4,
                    camera_settings={
                        "lens": faker.word(),
                    },
                ),
                SourceIn(
                    device_label=device_label,
                    file_id=faker.uuid4(),
                    format=TakeSourceKey.move,
                    camera_settings={
                        "lens": faker.word(),
                    },
                ),
            ],
            human_height=faker.pyfloat(positive=True),
            area_type=faker.enum(AreaType),
            name=faker.word(),
            expand=expand,
        )
        suffix = "_".join(expand) if expand else str(expand)
        self.assert_execute(
            snapshot,
            name=f"create_mutation_expand_{suffix}",
        )
        assert volume.model_dump(mode="json") == snapshot_json(
            name=f"create_response_expand_{suffix}",
        )

    @pytest.mark.parametrize(
        argnames="expand, take_fixture",
        argvalues=[
            (None, "volume_get_response"),
            ([], "volume_get_response"),
            (["client"], "volume_get_response_with_client"),
            ([OUTPUTS_LITERAL], "volume_get_response_with_outputs"),
            ([SOURCES_LITERAL], "volume_get_response_with_video_source"),
            ([SOURCES_LITERAL], "volume_get_res_with_additional_sources"),
        ],
        ids=[
            "no_expand",
            "empty_expand",
            "expand_client",
            "expand_outputs",
            "expand_video_file",
            "expand_additional_sources",
        ],
    )
    def test_retrieve(  # noqa: WPS211
        self,
        snapshot,
        snapshot_json,
        faker,
        request,
        expand,
        take_fixture,
    ):
        """Test creating a take.

        This should test -> `ugc.volumes.create_human_volume()`

        Args:
            snapshot: The snapshot fixture.
            snapshot_json: The snapshot json fixture.
            faker: The faker fixture.
            request: The request fixture.
            expand: The expand fixture.
            take_fixture: The take fixture.
        """
        request.getfixturevalue(take_fixture)
        volume = self.client.volumes.retrieve_human_volume(
            id=f"volume-{faker.uuid4()}",
            expand=expand,
        )
        suffix = "_".join(expand) if expand else str(expand)
        self.assert_execute(
            snapshot,
            name=f"get_query_expand_{suffix}",
        )
        assert volume.model_dump(mode="json") == snapshot_json(
            name=f"get_response_expand_{suffix}",
        )

    def test_list(self, snapshot, snapshot_json, volume_list_response):
        """Test listing volumes.

        This should test -> `ugc.volumes.list()`

        Args:
            snapshot: The snapshot fixture.
            snapshot_json: The snapshot json fixture.
            volume_list_response: volume list response fixture.
        """
        volume_list = self.client.volumes.list()
        self.assert_execute(snapshot, name="volume_list_request")
        assert volume_list.model_dump(mode="json") == snapshot_json(
            name="list_response",
        )

    def test_list_volume_invalid(self, request):
        """Test volume errors.

        This should test -> `ugc.volumes.list()`

        Args:
            request: The request fixture.
        """
        mock_transport = request.getfixturevalue("mock_transport")
        fake_list_volume_response = request.getfixturevalue("fake_list_volume_response")
        introspection_result = request.getfixturevalue("introspection_result")

        fake_list_volume_response[LIST_VOLUMES_QUERY]["items"] = [
            {"id": "dummy-123-123-123-123"},
        ]

        volume_response = ExecutionResult(data=fake_list_volume_response)
        mock_transport.side_effect = [introspection_result, volume_response]

        with pytest.raises(ValidationError) as excinfo:
            self.client.volumes.list()

    def test_list_volume_empty(self, request):
        """Test volume when it has an empty response.

        This should test -> `ugc.volumes.list()`

        Args:
            request: The request fixture.
        """
        mock_transport = request.getfixturevalue("mock_transport")
        fake_list_volume_response = request.getfixturevalue("fake_list_volume_response")
        introspection_result = request.getfixturevalue("introspection_result")

        fake_list_volume_response[LIST_VOLUMES_QUERY]["items"] = []

        volume_response = ExecutionResult(data=fake_list_volume_response)
        mock_transport.side_effect = [introspection_result, volume_response]

        assert not self.client.volumes.list().items
