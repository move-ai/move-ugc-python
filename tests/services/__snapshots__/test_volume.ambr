# serializer version: 1
# name: TestVolumeService.test_create[empty_expand][create_mutation_expand_[]]
  list([
    list([
      '''
        mutation create($sources: [SourceInput!], $syncMethod: SyncMethodInput, $areaType: AreaType!, $humanHeight: Float!, $metadata: AWSJSON, $name: String, $clip_window: ClipWindowInput) {
          createVolumeWithHuman(
            sources: $sources
            syncMethod: $syncMethod
            areaType: $areaType
            humanHeight: $humanHeight
            metadata: $metadata
            name: $name
            clipWindow: $clip_window
          ) {
            id
            areaType
            created
            humanHeight
            metadata
            name
            state
            progress {
              state
              percentageComplete
            }
            __typename
          }
        }
      ''',
    ]),
    dict({
      'operation_name': None,
      'variable_values': dict({
        'areaType': 'NORMAL',
        'clip_window': dict({
          'endTime': 9057.0,
          'startTime': 3757.0,
        }),
        'humanHeight': 873121848153.264,
        'metadata': '{"decade": "zAvJMvacZIYSmMsDUNvC", "other": "XbBPNrbhtJksbBuoWXSK", "draw": "PnbQcVNCliYtuFCSJkGb", "table": "ACMVZcKAiGBcYgCzHAad", "huge": "eFaLyHGEQQpSzpRFSYEm", "last": "dnZCcfgZNBnaEkbOzIyO", "trouble": "UgnhIyaDJzohUigyDYZf", "analysis": "UmKdTFlLMIuIvJkRJnoM", "house": "aYyOUXkJPUjPJGpDdakX", "director": "KNmpExWtgQLcAEuRyBkN"}',
        'name': 'attention',
        'sources': list([
          dict({
            'cameraSettings': dict({
              'lens': 'design',
            }),
            'deviceLabel': 'realize',
            'fileId': '530e9daa-0d45-45a7-a849-1a43874a50b9',
            'format': 'MP4',
          }),
          dict({
            'cameraSettings': dict({
              'lens': 'so',
            }),
            'deviceLabel': 'realize',
            'fileId': 'b898a7a5-37be-4600-bdc6-8ffa06c7a4f6',
            'format': 'MOVE',
          }),
        ]),
        'syncMethod': None,
      }),
    }),
  ])
# ---
# name: TestVolumeService.test_create[expand_additional_sources][create_mutation_expand_sources]
  list([
    list([
      '''
        mutation create($sources: [SourceInput!], $syncMethod: SyncMethodInput, $areaType: AreaType!, $humanHeight: Float!, $metadata: AWSJSON, $name: String, $clip_window: ClipWindowInput) {
          createVolumeWithHuman(
            sources: $sources
            syncMethod: $syncMethod
            areaType: $areaType
            humanHeight: $humanHeight
            metadata: $metadata
            name: $name
            clipWindow: $clip_window
          ) {
            id
            areaType
            created
            humanHeight
            metadata
            name
            state
            progress {
              state
              percentageComplete
            }
            sources {
              deviceLabel
              file {
                id
                created
                type
                metadata
                name
                presignedUrl
                thumbnailUrl
                __typename
              }
              cameraSettings {
                lens
              }
              format
            }
            __typename
          }
        }
      ''',
    ]),
    dict({
      'operation_name': None,
      'variable_values': dict({
        'areaType': 'NORMAL',
        'clip_window': dict({
          'endTime': 9057.0,
          'startTime': 3757.0,
        }),
        'humanHeight': 873121848153.264,
        'metadata': '{"decade": "zAvJMvacZIYSmMsDUNvC", "other": "XbBPNrbhtJksbBuoWXSK", "draw": "PnbQcVNCliYtuFCSJkGb", "table": "ACMVZcKAiGBcYgCzHAad", "huge": "eFaLyHGEQQpSzpRFSYEm", "last": "dnZCcfgZNBnaEkbOzIyO", "trouble": "UgnhIyaDJzohUigyDYZf", "analysis": "UmKdTFlLMIuIvJkRJnoM", "house": "aYyOUXkJPUjPJGpDdakX", "director": "KNmpExWtgQLcAEuRyBkN"}',
        'name': 'attention',
        'sources': list([
          dict({
            'cameraSettings': dict({
              'lens': 'design',
            }),
            'deviceLabel': 'realize',
            'fileId': '530e9daa-0d45-45a7-a849-1a43874a50b9',
            'format': 'MP4',
          }),
          dict({
            'cameraSettings': dict({
              'lens': 'so',
            }),
            'deviceLabel': 'realize',
            'fileId': 'b898a7a5-37be-4600-bdc6-8ffa06c7a4f6',
            'format': 'MOVE',
          }),
        ]),
        'syncMethod': None,
      }),
    }),
  ])
# ---
# name: TestVolumeService.test_create[expand_client][create_mutation_expand_client]
  list([
    list([
      '''
        mutation create($sources: [SourceInput!], $syncMethod: SyncMethodInput, $areaType: AreaType!, $humanHeight: Float!, $metadata: AWSJSON, $name: String, $clip_window: ClipWindowInput) {
          createVolumeWithHuman(
            sources: $sources
            syncMethod: $syncMethod
            areaType: $areaType
            humanHeight: $humanHeight
            metadata: $metadata
            name: $name
            clipWindow: $clip_window
          ) {
            id
            areaType
            created
            humanHeight
            metadata
            name
            state
            progress {
              state
              percentageComplete
            }
            client {
              id
              name
              created
              metadata
              portal
              __typename
            }
            __typename
          }
        }
      ''',
    ]),
    dict({
      'operation_name': None,
      'variable_values': dict({
        'areaType': 'NORMAL',
        'clip_window': dict({
          'endTime': 9057.0,
          'startTime': 3757.0,
        }),
        'humanHeight': 873121848153.264,
        'metadata': '{"decade": "zAvJMvacZIYSmMsDUNvC", "other": "XbBPNrbhtJksbBuoWXSK", "draw": "PnbQcVNCliYtuFCSJkGb", "table": "ACMVZcKAiGBcYgCzHAad", "huge": "eFaLyHGEQQpSzpRFSYEm", "last": "dnZCcfgZNBnaEkbOzIyO", "trouble": "UgnhIyaDJzohUigyDYZf", "analysis": "UmKdTFlLMIuIvJkRJnoM", "house": "aYyOUXkJPUjPJGpDdakX", "director": "KNmpExWtgQLcAEuRyBkN"}',
        'name': 'attention',
        'sources': list([
          dict({
            'cameraSettings': dict({
              'lens': 'design',
            }),
            'deviceLabel': 'realize',
            'fileId': '530e9daa-0d45-45a7-a849-1a43874a50b9',
            'format': 'MP4',
          }),
          dict({
            'cameraSettings': dict({
              'lens': 'so',
            }),
            'deviceLabel': 'realize',
            'fileId': 'b898a7a5-37be-4600-bdc6-8ffa06c7a4f6',
            'format': 'MOVE',
          }),
        ]),
        'syncMethod': None,
      }),
    }),
  ])
# ---
# name: TestVolumeService.test_create[expand_clip_window][create_mutation_expand_clipWindow]
  list([
    list([
      '''
        mutation create($sources: [SourceInput!], $syncMethod: SyncMethodInput, $areaType: AreaType!, $humanHeight: Float!, $metadata: AWSJSON, $name: String, $clip_window: ClipWindowInput) {
          createVolumeWithHuman(
            sources: $sources
            syncMethod: $syncMethod
            areaType: $areaType
            humanHeight: $humanHeight
            metadata: $metadata
            name: $name
            clipWindow: $clip_window
          ) {
            id
            areaType
            created
            humanHeight
            metadata
            name
            state
            progress {
              state
              percentageComplete
            }
            clipWindow {
              startTime
              endTime
            }
            __typename
          }
        }
      ''',
    ]),
    dict({
      'operation_name': None,
      'variable_values': dict({
        'areaType': 'LARGE',
        'clip_window': dict({
          'endTime': 5315.0,
          'startTime': 849.0,
        }),
        'humanHeight': 59816290182492.2,
        'metadata': '{"decade": "zAvJMvacZIYSmMsDUNvC", "other": "XbBPNrbhtJksbBuoWXSK", "draw": "PnbQcVNCliYtuFCSJkGb", "table": "ACMVZcKAiGBcYgCzHAad", "huge": "eFaLyHGEQQpSzpRFSYEm", "last": "dnZCcfgZNBnaEkbOzIyO", "trouble": "UgnhIyaDJzohUigyDYZf", "analysis": "UmKdTFlLMIuIvJkRJnoM", "house": "aYyOUXkJPUjPJGpDdakX", "director": "KNmpExWtgQLcAEuRyBkN"}',
        'name': 'draw',
        'sources': list([
          dict({
            'cameraSettings': dict({
              'lens': 'set',
            }),
            'deviceLabel': 'newspaper',
            'fileId': '37bea600-bdc6-4ffa-86c7-a4f634c4614f',
            'format': 'MP4',
          }),
          dict({
            'cameraSettings': dict({
              'lens': 'evening',
            }),
            'deviceLabel': 'newspaper',
            'fileId': 'cba1d785-4a1e-4b59-aca4-d775bfdde401',
            'format': 'MOVE',
          }),
        ]),
        'syncMethod': None,
      }),
    }),
  ])
# ---
# name: TestVolumeService.test_create[expand_video_file][create_mutation_expand_sources]
  list([
    list([
      '''
        mutation create($sources: [SourceInput!], $syncMethod: SyncMethodInput, $areaType: AreaType!, $humanHeight: Float!, $metadata: AWSJSON, $name: String, $clip_window: ClipWindowInput) {
          createVolumeWithHuman(
            sources: $sources
            syncMethod: $syncMethod
            areaType: $areaType
            humanHeight: $humanHeight
            metadata: $metadata
            name: $name
            clipWindow: $clip_window
          ) {
            id
            areaType
            created
            humanHeight
            metadata
            name
            state
            progress {
              state
              percentageComplete
            }
            sources {
              deviceLabel
              file {
                id
                created
                type
                metadata
                name
                presignedUrl
                thumbnailUrl
                __typename
              }
              cameraSettings {
                lens
              }
              format
            }
            __typename
          }
        }
      ''',
    ]),
    dict({
      'operation_name': None,
      'variable_values': dict({
        'areaType': 'NORMAL',
        'clip_window': dict({
          'endTime': 9057.0,
          'startTime': 3757.0,
        }),
        'humanHeight': 873121848153.264,
        'metadata': '{"decade": "zAvJMvacZIYSmMsDUNvC", "other": "XbBPNrbhtJksbBuoWXSK", "draw": "PnbQcVNCliYtuFCSJkGb", "table": "ACMVZcKAiGBcYgCzHAad", "huge": "eFaLyHGEQQpSzpRFSYEm", "last": "dnZCcfgZNBnaEkbOzIyO", "trouble": "UgnhIyaDJzohUigyDYZf", "analysis": "UmKdTFlLMIuIvJkRJnoM", "house": "aYyOUXkJPUjPJGpDdakX", "director": "KNmpExWtgQLcAEuRyBkN"}',
        'name': 'attention',
        'sources': list([
          dict({
            'cameraSettings': dict({
              'lens': 'design',
            }),
            'deviceLabel': 'realize',
            'fileId': '530e9daa-0d45-45a7-a849-1a43874a50b9',
            'format': 'MP4',
          }),
          dict({
            'cameraSettings': dict({
              'lens': 'so',
            }),
            'deviceLabel': 'realize',
            'fileId': 'b898a7a5-37be-4600-bdc6-8ffa06c7a4f6',
            'format': 'MOVE',
          }),
        ]),
        'syncMethod': None,
      }),
    }),
  ])
# ---
# name: TestVolumeService.test_create[no_expand][create_mutation_expand_None]
  list([
    list([
      '''
        mutation create($sources: [SourceInput!], $syncMethod: SyncMethodInput, $areaType: AreaType!, $humanHeight: Float!, $metadata: AWSJSON, $name: String, $clip_window: ClipWindowInput) {
          createVolumeWithHuman(
            sources: $sources
            syncMethod: $syncMethod
            areaType: $areaType
            humanHeight: $humanHeight
            metadata: $metadata
            name: $name
            clipWindow: $clip_window
          ) {
            id
            areaType
            created
            humanHeight
            metadata
            name
            state
            progress {
              state
              percentageComplete
            }
            __typename
          }
        }
      ''',
    ]),
    dict({
      'operation_name': None,
      'variable_values': dict({
        'areaType': 'NORMAL',
        'clip_window': dict({
          'endTime': 9057.0,
          'startTime': 3757.0,
        }),
        'humanHeight': 873121848153.264,
        'metadata': '{"decade": "zAvJMvacZIYSmMsDUNvC", "other": "XbBPNrbhtJksbBuoWXSK", "draw": "PnbQcVNCliYtuFCSJkGb", "table": "ACMVZcKAiGBcYgCzHAad", "huge": "eFaLyHGEQQpSzpRFSYEm", "last": "dnZCcfgZNBnaEkbOzIyO", "trouble": "UgnhIyaDJzohUigyDYZf", "analysis": "UmKdTFlLMIuIvJkRJnoM", "house": "aYyOUXkJPUjPJGpDdakX", "director": "KNmpExWtgQLcAEuRyBkN"}',
        'name': 'attention',
        'sources': list([
          dict({
            'cameraSettings': dict({
              'lens': 'design',
            }),
            'deviceLabel': 'realize',
            'fileId': '530e9daa-0d45-45a7-a849-1a43874a50b9',
            'format': 'MP4',
          }),
          dict({
            'cameraSettings': dict({
              'lens': 'so',
            }),
            'deviceLabel': 'realize',
            'fileId': 'b898a7a5-37be-4600-bdc6-8ffa06c7a4f6',
            'format': 'MOVE',
          }),
        ]),
        'syncMethod': None,
      }),
    }),
  ])
# ---
# name: TestVolumeService.test_fetch_service
  VolumeService(api_key=SecretStr('**********'), endpoint_url=HttpUrl('https://pytest_invalid_endpoint_url.com/'))
# ---
# name: TestVolumeService.test_list[volume_list_request]
  list([
    list([
      '''
        query list($first: Int, $after: AWSJSON, $sortDirection: SortDirection) {
          listVolumes(first: $first, after: $after, sortDirection: $sortDirection) {
            first
            after
            items {
              ... on Volume {
                ...VolumeFields
              }
            }
          }
        }
        
        fragment VolumeFields on HumanVolume {
          id
          areaType
          created
          humanHeight
          metadata
          name
          state
          progress {
            state
            percentageComplete
          }
          __typename
        }
      ''',
    ]),
    dict({
      'operation_name': None,
      'variable_values': dict({
        'after': None,
        'expand': None,
        'first': 10,
        'sortDirection': 'DESC',
        'takeId': None,
      }),
    }),
  ])
# ---
# name: TestVolumeService.test_retrieve[empty_expand][get_query_expand_[]]
  list([
    list([
      '''
        query retrieve($id: ID!) {
          getVolume(id: $id) {
            ... on Volume {
              ...VolumeFields
            }
          }
        }
        
        fragment VolumeFields on HumanVolume {
          id
          areaType
          created
          humanHeight
          metadata
          name
          state
          progress {
            state
            percentageComplete
          }
          __typename
        }
      ''',
    ]),
    dict({
      'operation_name': None,
      'variable_values': dict({
        'id': 'volume-8d866324-3ab5-4d05-a9c5-e560a8020f0a',
      }),
    }),
  ])
# ---
# name: TestVolumeService.test_retrieve[expand_additional_sources][get_query_expand_sources]
  list([
    list([
      '''
        query retrieve($id: ID!) {
          getVolume(id: $id) {
            ... on Volume {
              ...VolumeFields
            }
          }
        }
        
        fragment VolumeFields on HumanVolume {
          id
          areaType
          created
          humanHeight
          metadata
          name
          state
          progress {
            state
            percentageComplete
          }
          sources {
            deviceLabel
            file {
              id
              created
              type
              metadata
              name
              presignedUrl
              thumbnailUrl
              __typename
            }
            cameraSettings {
              lens
            }
            format
          }
          __typename
        }
      ''',
    ]),
    dict({
      'operation_name': None,
      'variable_values': dict({
        'id': 'volume-874a50b9-8d86-4324-bab5-ed05e9c5e560',
      }),
    }),
  ])
# ---
# name: TestVolumeService.test_retrieve[expand_client][get_query_expand_client]
  list([
    list([
      '''
        query retrieve($id: ID!) {
          getVolume(id: $id) {
            ... on Volume {
              ...VolumeFields
            }
          }
        }
        
        fragment VolumeFields on HumanVolume {
          id
          areaType
          created
          humanHeight
          metadata
          name
          state
          progress {
            state
            percentageComplete
          }
          client {
            id
            name
            created
            metadata
            portal
            __typename
          }
          __typename
        }
      ''',
    ]),
    dict({
      'operation_name': None,
      'variable_values': dict({
        'id': 'volume-8d866324-3ab5-4d05-a9c5-e560a8020f0a',
      }),
    }),
  ])
# ---
# name: TestVolumeService.test_retrieve[expand_outputs][get_query_expand_outputs]
  list([
    list([
      '''
        query retrieve($id: ID!) {
          getVolume(id: $id) {
            ... on Volume {
              ...VolumeFields
            }
          }
        }
        
        fragment VolumeFields on HumanVolume {
          id
          areaType
          created
          humanHeight
          metadata
          name
          state
          progress {
            state
            percentageComplete
          }
          outputs {
            format
            file {
              id
              created
              type
              metadata
              name
              presignedUrl
              thumbnailUrl
              __typename
            }
            __typename
          }
          __typename
        }
      ''',
    ]),
    dict({
      'operation_name': None,
      'variable_values': dict({
        'id': 'volume-8d866324-3ab5-4d05-a9c5-e560a8020f0a',
      }),
    }),
  ])
# ---
# name: TestVolumeService.test_retrieve[expand_video_file][get_query_expand_sources]
  list([
    list([
      '''
        query retrieve($id: ID!) {
          getVolume(id: $id) {
            ... on Volume {
              ...VolumeFields
            }
          }
        }
        
        fragment VolumeFields on HumanVolume {
          id
          areaType
          created
          humanHeight
          metadata
          name
          state
          progress {
            state
            percentageComplete
          }
          sources {
            deviceLabel
            file {
              id
              created
              type
              metadata
              name
              presignedUrl
              thumbnailUrl
              __typename
            }
            cameraSettings {
              lens
            }
            format
          }
          __typename
        }
      ''',
    ]),
    dict({
      'operation_name': None,
      'variable_values': dict({
        'id': 'volume-874a50b9-8d86-4324-bab5-ed05e9c5e560',
      }),
    }),
  ])
# ---
# name: TestVolumeService.test_retrieve[no_expand][get_query_expand_None]
  list([
    list([
      '''
        query retrieve($id: ID!) {
          getVolume(id: $id) {
            ... on Volume {
              ...VolumeFields
            }
          }
        }
        
        fragment VolumeFields on HumanVolume {
          id
          areaType
          created
          humanHeight
          metadata
          name
          state
          progress {
            state
            percentageComplete
          }
          __typename
        }
      ''',
    ]),
    dict({
      'operation_name': None,
      'variable_values': dict({
        'id': 'volume-8d866324-3ab5-4d05-a9c5-e560a8020f0a',
      }),
    }),
  ])
# ---
